packages/lib-classifier/src/store/SubjectStore/SubjectStore.js

```diff
@@ -176,7 +182,18 @@ const SubjectStore = types
       newSubjects.forEach(subject => {
         try {
           const alreadyStored = self.resources.get(subject.id)
-          if (!alreadyStored) {
+          // assume all subjects from Panoptes are unseen by default
+          let notSeen = true
+          // completed workflows fall back to random selection
+          const { user_has_finished_workflow } = subject
```

Comment:
OK, quick sanity check. user_has_finished_workflow is taken straight from the API... so is there any scenario that a Subject says "user hasn't finished workflow?" but the user actually has JUST finished the WF (e.g. they're submitting the classification for the final Subject now!) so the API hasn't updated that boolean yet?

Probably? Maybe? But it's such a narrow edge case I'm having trouble coming up with a test for it.

Reply:
I suppose if they were one subject away from the end, and this is the subject that completes the workflow, but I'm getting tired of adding guard clauses to the classifier, to be honest.